File: src/main/java/io/github/aphelium/modules/coin/dtos/CreateCryptoDTO.java
package io.github.aphelium.modules.coin.dtos;

import lombok.Data;

import javax.validation.constraints.NotBlank;

@Data
public class CreateCryptoDTO {

    @NotBlank(message = "The name is mandatory")
    private String name;
    @NotBlank(message = "The description is mandatory")
    private String description;
    @NotBlank(message = "The code is mandatory")
    private String code;
    private Boolean isActive;
    private String logo;
    @NotBlank(message = "The first historical date is mandatory")
    private String firstHistoricalDate;
    private String lastHistoricalDate;


}

File: src/main/kotlin/io/github/aphelium/modules/coin/dtos/CreateCryptoDTO.kt
package io.github.aphelium.modules.coin.dtos

import javax.validation.constraints.NotBlank

class CreateCryptoDTO {
    val name: @NotBlank(message = "The name is mandatory") String? = null
    val description: @NotBlank(message = "The description is mandatory") String? = null
    val code: @NotBlank(message = "The code is mandatory") String? = null
    val isActive: Boolean? = null
    val logo: String? = null
    val firstHistoricalDate: @NotBlank(message = "The first historical date is mandatory") String? = null
    val lastHistoricalDate: String? = null
}